presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -80,6 +81,7 @@
     private final Map<String, Map<String, String>> unprocessedCatalogProperties;
     private final SessionPropertyManager sessionPropertyManager;
     private final Map<String, String> preparedStatements;
+    private final AccessControlContext accessControlContext;
```

Comment:
Wouldn't it be bad to create a new AccessControlContext object everytime we need it ?
Instead can we NOT have it as an input param to the Session object constructor but keep it as private variable in Session object and create/initialize it in the constructor  ?

Reply:
> Wouldn't it be bad to create a new AccessControlContext object everytime we need it ?

Creating an object is cheap. Especially if it is very likely to get in-lined as a result of escape analysis. Also there's not that many checks that are performed per query. Thus I would consider additional overhead of creating an object to be negligible. Also in many cases (for example at workers) no security checks are done. So the object is not going to be created altogether. 

> Instead can we NOT have it as an input param to the Session object constructor but keep it as private variable in Session object and create/initialize it inside the constructor method ?

Yeah, we can create it in the constructor based on the parameters. It will protect the `Session` object from being in an inconsistent state (e.g.: when queryId of the `Session` is different then queryId of the `AccessControlContext`)

Reply:
Sounds good. Will go ahead with the idea of having ``AccessControlContext``  as private variable in ``Session`` and creating/initializing it as part of the ``Session`` constructor.